src/main/java/io/weaviate/client/v1/async/schema/api/SchemaDeleter.java

```diff
@@ -0,0 +1,79 @@
+package io.weaviate.client.v1.async.schema.api;
+
+import io.weaviate.client.base.AsyncClientResult;
+import io.weaviate.client.base.Result;
+import io.weaviate.client.base.WeaviateErrorMessage;
+import io.weaviate.client.base.WeaviateErrorResponse;
+import io.weaviate.client.v1.schema.model.Schema;
+import io.weaviate.client.v1.schema.model.WeaviateClass;
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionException;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.Future;
+import java.util.stream.Collectors;
+import org.apache.hc.core5.concurrent.FutureCallback;
+
+public class SchemaDeleter implements AsyncClientResult<Boolean> {
+
+  private final SchemaGetter schemaGetter;
+  private final ClassDeleter classDeleter;
+
+  public SchemaDeleter(SchemaGetter schemaGetter, ClassDeleter classDeleter) {
+    this.schemaGetter = schemaGetter;
+    this.classDeleter = classDeleter;
+  }
+
+  @Override
+  public Future<Result<Boolean>> run() {
+    return deleteAll(null);
+  }
+
+  @Override
+  public Future<Result<Boolean>> run(FutureCallback<Result<Boolean>> callback) {
+    return deleteAll(callback);
+  }
+
+  private Future<Result<Boolean>> deleteAll(FutureCallback<Result<Boolean>> callback) {
+    CompletableFuture<Result<Schema>> getSchema = CompletableFuture.supplyAsync(() -> {
+      try {
+        return schemaGetter.run().get();
+      } catch (InterruptedException | ExecutionException e) {
+        throw new CompletionException(e);
+      }
+    });
+    CompletableFuture<Result<Boolean>> deleteAll = getSchema.thenApplyAsync(schema -> {
+      if (schema.getError() != null) {
+        List<WeaviateErrorMessage> errorMessages = schema.getError().getMessages().stream().map(err ->
+          WeaviateErrorMessage.builder().message(err.getMessage()).build()
+        ).collect(Collectors.toList());
+        WeaviateErrorResponse errors = WeaviateErrorResponse.builder()
+          .error(errorMessages).build();
+        return new Result<>(schema.getError().getStatusCode(), false, errors);
+      } else {
+        List<WeaviateClass> weaviateClasses = schema.getResult().getClasses();
+        for (WeaviateClass clazz : weaviateClasses) {
+          try {
+            Result<Boolean> result = classDeleter.withClassName(clazz.getClassName()).run().get();
+            if (result.getError() != null) {
+              return result;
+            }
+          } catch (InterruptedException | ExecutionException e) {
+            throw new CompletionException(e);
+          }
+        }
```

Comment:
Nit: `return` right before `else` -- we can shave off some indentation here by moving the else-brach code out of the else branch

Another thing I thought about here: couldn't we continue using  CompletableFutures in the "then"-callback to have each of "deletes" run asynchronously and then combine them with `CompletableFuture.allOf()`?

If yes, it would probably also make sense to switch from `thenApplyAsync` to `thenApply`, since the "deletes" will run in their individual threads anyways.

Reply:
I was actually thinking of `CompletableFuture.allOf()` and I had an implementation for this already done, but then I thought that we would have to limit the number of threads here and that we would have to add executor service here and set it to some max number of threads and then I thought that we would have to also find a way of cancelling the threads if one resulted in error and then finally I thought that I will do this in a separate PR if needed :) and just copy the current impl here

Reply:
Yeah you're right, I see there are some strings attach to that approach and it's best not to bloat the PR with solving them too 👍